### PR TITLE
Add text for RFC90: Direct access to compressed raster data

### PR DESCRIPTION
**Pull request by rouault**
Rendered view [here](https://github.com/rouault/gdal/blob/rfc90_text/doc/source/development/rfc/rfc90_read_compressed_data.rst)

### CODE REVIEW

**Comment by hyoklee** [comment]
The idea in this RFC sounds fantastic! I think it will boost the GDAL performance significantly.

**Comment by hyoklee** [comment]
Please fix a typo *capabities*:

`a candidate implementation with the following capabities:`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+The scenarios covered by the below proposed implementation will be tested
+by C++ unit tests (unit testing of GetCompressionFormats() and ReadCompressedData()
+implementations), and in Python autotest suite for end-to-end tests (e.g lossless
+JPEG -> JPEGXL -> JPEG)
+
+Issues / pull requests
+----------------------
+
+https://github.com/OSGeo/gdal/compare/master...rouault:gdal:ReadCompressedData?expand=1 contains
+a candidate implementation with the following capabities:
```
Comment (rouault) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:414]:
```suggestion
a candidate implementation with the following capabilities:
```

**Comment by rouault** [comment]
> Please fix a typo capabities:

done. Note: you can use the comment functionality of github to directly suggest edits.

**Comment by mdsumner** [comment]
`be example` occurs twice in code comments, I assume it should be `be able`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
++++++++++++++++++++++++
+
+.. code-block:: c++
+
+    /** Return the compression formats that can be natively obtained for the
+     * window of interest and requested bands.
+     *
+     * For example, a tiled dataset may be able to return data in a compressed
+     * format if the window of interest matches exactly a tile. For some formats,
+     * drivers may also be example to merge several tiles together (not currently
```
Comment (mdsumner) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:108]: `be able to`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+++++++++++++++++++++
+
+.. code-block:: c++
+
+    /** Return the compressed content that can be natively obtained for the
+     * window of interest and requested bands.
+     *
+     * For example, a tiled dataset may be able to return data in compressed format
+     * is the window of interest matches exactly a tile. For some formats, drivers
+     * may also be example to merge several tiles together (not currently
```
Comment (mdsumner) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:188]: `be able to`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+Users might also set it to NO to ask for the regular code path to be taken.
+Setting it to NO should be uncommon. This is a provision in case the
+optimized code path would have bugs, or if for any other reason, the regular
+code path must be taken (if the source compressed data was not fully conformant
+for example).
+
+Miscellaneous
++++++++++++++
+
+An helper ``bool GDALDataset::IsAllBands(int nBandCount, const int *panBandList) const``
```
Comment (mdsumner) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:363]: `A helper`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+     * window of interest and requested bands.
+     *
+     * For example, a tiled dataset may be able to return data in compressed format
+     * is the window of interest matches exactly a tile. For some formats, drivers
+     * may also be example to merge several tiles together (not currently
+     * implemented though).
+     *
+     * The implementation should make sure that the content returned forms a valid
+     * standalone file. For example, for the GeoTIFF implementation of this method,
+     * when extracting a JPEG tile, the method will automatically adds the content
```
Comment (mdsumner) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:193]: `automatically add the`

File: doc/source/development/rfc/rfc90_read_compressed_data.rst (hunk 4 above)
Comment (rouault) [reply doc/source/development/rfc/rfc90_read_compressed_data.rst]: fixed. thanks

File: doc/source/development/rfc/rfc90_read_compressed_data.rst (hunk 5 above)
Comment (rouault) [reply doc/source/development/rfc/rfc90_read_compressed_data.rst]: fixed. thanks

File: doc/source/development/rfc/rfc90_read_compressed_data.rst [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+
+- gdal_translate of a JPEGXL file with JPEG reconstruction box to JPEG
+  (involves the JPEGXL driver as producer, the JPEG driver as consumer). And
+  the reverse operation: lossless conversion of JPEG to JPEGXL with a JPEG
+  reconstruction box.
+
+Voting history
+--------------
+
+TODO
```
Comment (rouault) [inline doc/source/development/rfc/rfc90_read_compressed_data.rst:458]:
```suggestion
+1 from PSC members JukkaR, FrankW, SeanG, MateuzL and EvenR
```

**Pull request merged**